src/Microsoft.CodeQuality.Analyzers/Core/QualityGuidelines/MarkMembersAsStatic.Fixer.cs

```diff
@@ -12,19 +22,185 @@ namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var symbol = semanticModel.GetDeclaredSymbol(node, cancellationToken);
+            if (symbol != null)
+            {
+                solution = await UpdateReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+            }
+
+            return solution;
+        }
+
+        private async Task<Solution> UpdateReferencesAsync(ISymbol symbol, Solution solution, CancellationToken cancellationToken)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var references = await SymbolFinder.FindReferencesAsync(symbol, solution, cancellationToken).ConfigureAwait(false);
+            if (references?.Count() != 1)
+            {
+                return solution;
+            }
+
+            foreach (var referenceLocationGroup in references.Single().Locations.GroupBy(r => r.Document))
+            {
+                // Get document in current solution
+                var document = solution.GetDocument(referenceLocationGroup.Key.Id);
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                // Compute replacements
+                var editor = new SyntaxEditor(root, solution.Workspace);
+                foreach (var referenceLocation in referenceLocationGroup)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    var referenceNode = root.FindNode(referenceLocation.Location.SourceSpan, getInnermostNodeForTie: true);
+                    if (referenceNode != null)
+                    {
+                        var operation = GetOperationWalkingUpParentChain(referenceNode, semanticModel);
+                        SyntaxNode nodeToReplaceOpt = null;
+                        switch (operation)
+                        {
+                            case IMemberReferenceOperation memberReference:
+                                if (IsReplacableOperation(memberReference.Instance))
+                                {
+                                    nodeToReplaceOpt = GetSyntaxNodeToReplace(memberReference);
+                                }
+
+                                break;
 
+                            case IInvocationOperation invocation:
+                                if (IsReplacableOperation(invocation.Instance))
+                                {
+                                    nodeToReplaceOpt = GetExpressionOfInvocation(invocation.Syntax);
+                                }
+
+                                break;
+                        }
+
+                        if (nodeToReplaceOpt != null)
```

Comment:
It will be if the call to `IsReplacableOperation` above returns false, i.e. say the original invocation of this method had an invocation or cast in its receiver, we don't remove that code as it might change semantics and instead just skip replacement.